db/db.go

```diff
@@ -188,6 +208,73 @@ func (db *DB) RevokeSSH(rci *RevokedCertificateInfo) error {
 	}
 }
 
+// GetRevokedCertificates gets a list of all revoked certificates.
+func (db *DB) GetRevokedCertificates() (*[]RevokedCertificateInfo, error) {
+	entries, err := db.List(revokedCertsTable)
+	if err != nil {
+		return nil, err
+	}
+	var revokedCerts []RevokedCertificateInfo
+	for _, e := range entries {
+		var data RevokedCertificateInfo
+		if err := json.Unmarshal(e.Value, &data); err != nil {
+			return nil, err
+		}
+
+		if !data.ExpiresAt.IsZero() && data.ExpiresAt.After(time.Now().UTC()) {
+			revokedCerts = append(revokedCerts, data)
+		} else if data.ExpiresAt.IsZero() {
+			cert, err := db.GetCertificate(data.Serial)
+			if err != nil {
+				revokedCerts = append(revokedCerts, data) // a revoked certificate may not be in the database,
+				// so its expiry date is undiscoverable and will need
+				// to be added to the crl always
+				continue
+			}
+
+			if cert.NotAfter.After(time.Now().UTC()) {
+				revokedCerts = append(revokedCerts, data)
+			}
+		}
+
+	}
+	return &revokedCerts, nil
+}
+
+// StoreCRL stores a CRL in the DB
+func (db *DB) StoreCRL(crlInfo *CertificateRevocationListInfo) error {
```

Comment:
The authority id can also be a field in the DB type, as it won't change. And the interface will remain as it is.

Reply:
im unsure how DB can access the configured AuthorityID without altering either the DB config, or the interface? this could be because i am looking at an older branch, but DB does not currently have access to the AuthorityConfig as far as i can see...?